cmd/gossamer/configcmd.go

```diff
@@ -0,0 +1,122 @@
+	"os"
+	"strings"
+	"github.com/ChainSafe/gossamer/cmd/utils"
+	cfg "github.com/ChainSafe/gossamer/config"
+	"github.com/ChainSafe/gossamer/goss"
+	"github.com/ChainSafe/gossamer/polkadb"
+	log "github.com/inconshreveable/log15"
+	"github.com/naoina/toml"
+	"github.com/urfave/cli"
+)
+var (
+	configFileFlag = cli.StringFlag{
+		Name:  "config",
+		Usage: "TOML configuration file",
+	}
+)
+// makeNode sets up node; opening badgerDB instance and returning the Goss container
+func makeNode(ctx *cli.Context) (*goss.Goss, error) {
+	fig, err := setConfig(ctx)
+	if err != nil {
+		log.Error("unable to extract required config", "err", err)
+	}
+	srv := setP2PConfig(ctx, fig.ServiceConfig)
+	datadir := setDatabaseDir(ctx, fig)
+	db, err := polkadb.NewBadgerDB(datadir)
+	if err != nil {
+		fmt.Println(err)
+	}
+	return &goss.Goss{
+		ServerConfig: fig.ServiceConfig,
+		Server:       srv,
+		Polkadb:      db,
+	}, nil
+}
+// setConfig checks for config.toml if --config flag is specified
+func setConfig(ctx *cli.Context) (*cfg.Config, error) {
+	var fig *cfg.Config
+	// Load config file.
+	if file := ctx.GlobalString(configFileFlag.Name); file != "" {
+		config, err := loadConfig(file)
+		if err != nil {
+			log.Warn("err loading toml file", "err", err.Error())
+			return fig, err
+		}
+		return config, nil
+	}
+	return fig, nil
+}
+// setDatabaseDir initializes directory for BadgerDB logs
+func setDatabaseDir(ctx *cli.Context, cfg *cfg.Config) string {
+	if cfg.BadgerDB.Datadir != "" {
+		return cfg.BadgerDB.Datadir
+	} else if file := ctx.GlobalString(utils.DataDirFlag.Name); file != "" {
+		return file
+	} else {
+		log.Error("must specify data directory")
+		return ""
+	}
+}
+// loadConfig loads the contents from config.toml and inits Config object
+func loadConfig(file string) (*cfg.Config, error) {
+	f, err := os.Open(file)
+	if err != nil {
+		panic(err)
+	}
+	defer func() {
+		err = f.Close()
+		if err != nil {
+			log.Warn("err closing conn", "err", err.Error())
+		}
+	}()
+	var config *cfg.Config
+	if err = toml.NewDecoder(f).Decode(&config); err != nil {
+		log.Error("decoding toml error", "err", err.Error())
+	}
+	return config, err
+}
+// setBootstrapNodes creates a list of bootstrap nodes from the command line
+// flags, reverting to pre-configured ones if none have been specified.
+func setBootstrapNodes(ctx *cli.Context, cfg *p2p.ServiceConfig) {
+	var urls []string
+	switch {
+	case ctx.GlobalIsSet(utils.BootnodesFlag.Name):
+		urls = strings.Split(ctx.GlobalString(utils.BootnodesFlag.Name), ",")
+	case cfg.BootstrapNodes != nil:
+		return // already set, don't apply defaults.
+	}
+	cfg.BootstrapNodes = append(cfg.BootstrapNodes, urls...)
+}
+// SetP2PConfig sets up the configurations required for P2P service
+func setP2PConfig(ctx *cli.Context, cfg *p2p.ServiceConfig) *p2p.Service {
+	setBootstrapNodes(ctx, cfg)
+	srv := startP2PService(cfg)
+	return srv
+}
+// startP2PService starts a p2p network layer from provided config
```

Comment:
should we have spaces between the functions? not a huge deal just for readability

Reply:
Yes, we definitely should!